htdocs/api/class/api_documents.class.php

```diff
@@ -547,21 +549,9 @@ public function getDocumentsListByElement($modulepart, $id = 0, $ref = '', $sort
 
 			$upload_dir = $conf->categorie->multidir_output[$object->entity].'/'.get_exdir($object->id, 2, 0, 0, $object, 'category').$object->id."/photos/".dol_sanitizeFileName($object->ref);
 		} elseif ($modulepart == 'ecm') {
-			throw new RestException(500, 'Modulepart Ecm not implemented yet.');
-			// require_once DOL_DOCUMENT_ROOT.'/ecm/class/ecmdirectory.class.php';
-
-			// if (!DolibarrApiAccess::$user->hasRight('ecm', 'read')) {
-			// 	throw new RestException(403);
-			// }
-
-			// // $object = new EcmDirectory($this->db);
-			// // $result = $object->fetch($ref);
-			// // if (!$result) {
-			// // 	throw new RestException(404, 'EcmDirectory not found');
-			// // }
-			// $upload_dir = $conf->ecm->dir_output;
-			// $type = 'all';
-			// $recursive = 0;
+			$upload_dir = $conf->ecm->dir_output . "/". $ref;
```

Comment:
@defrance what happen if ref is ../../../mydir

Reply:
you call it with a valid token and the good rights so ... open bar

Reply:
another important point, we are here on the procedure which only lists the files present in the directory. The real question is more on the download and the upload that we should ask ourselves...

Reply:
You can (must) use dol_sanitizePathName($ref) so using .. in path is no more possible.

Reply:
you mean dol_sanitizeFileName($ref) instead
i made change in this way